lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -123,10 +154,13 @@ def relationship_value_for(serializer, options = {})
         end
 
         def relationships_for(serializer)
-          Hash[serializer.associations.map { |association| [association.key, { data: relationship_value_for(association.serializer, association.options) }] }]
+          Hash[serializer.associations.map { |association|
+            [association.key, { data: relationship_value_for(association.serializer, association.options) }]
+          }]
         end
 
         def included_for(serializer)
+          included = ActiveModel::Serializer::Utils.include_args_to_hash(instance_options[:include])
```

Comment:
This will recompute the hash each time `serializable_hash` gets called. Granted, currently it should only happen once, but there might be outside-Rails use-cases where it's not the case.